core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -300,6 +349,22 @@ private[spark] class SparkSubmitArguments(args: Seq[String], env: Map[String, St
         propertiesFile = value
         parse(tail)
 
+      case ("--kill") :: value :: tail =>
```

Comment:
this method for having actions with a flag is a bit akward... the Hadoop scripts did a better job with this by having the first word be the required action. Perhaps we should do that at some point (this approach leaves open that possibility).